commitizen/git.py

```diff
@@ -82,13 +82,16 @@ def from_line(cls, line: str, inner_delimiter: str) -> GitTag:
         return cls(name=name, rev=obj, date=date)
 
 
-def tag(tag: str, annotated: bool = False, signed: bool = False) -> cmd.Command:
+def tag(
+    tag: str, annotated: bool = False, signed: bool = False, msg: str | None = None
+) -> cmd.Command:
     _opt = ""
     if annotated:
         _opt = f"-a {tag} -m"
     if signed:
         _opt = f"-s {tag} -m"
-    c = cmd.run(f"git tag {_opt} {tag}")
+
+    c = cmd.run(f'git tag {_opt} "{tag if _opt == "" or msg is None else msg}"')
```

Comment:
@woile do you mean we should add a warning here to let the users know if they're using this feature, they're actually creating annotated tag?

>  To create a lightweight tag, don’t supply any of the -a, -s, or -m options, just provide a tag name:

https://git-scm.com/book/en/v2/Git-Basics-Tagging

not sure whether my understanding is correct

Reply:
I see, maybe a comment there explaining that. I didn't know that if you provide a message it automatically becomes an annotated tag.

Reply:
@woile I just added the comment and resolved the conflict. Could you please take a look? I think we're pretty close to merge this one :)

cc @noirbizarre